packages/pangea-sdk/pangea/services/file_scan.py

```diff
@@ -117,7 +122,15 @@ def file_scan(
                     print(f"\\t{err.detail} \\n")
         """
 
-        if file or file_path:
+        if transfer_method == TransferMethod.SOURCE_URL and source_url is None:
+            raise ValueError("`source_url` argument is required when using `TransferMethod.SOURCE_URL`.")
+
+        if source_url is not None and transfer_method != TransferMethod.SOURCE_URL:
+            raise ValueError(
+                "`transfer_method` should be `TransferMethod.SOURCE_URL` when using the `source_url` argument."
+            )
+
+        if file or file_path or source_url:
```

Comment:
Checking `source_url` shouldn't go in the `elif` block? I mean, we don't want to create the `files: List[Tuple] = [("upload", ("filename", file, "application/octet-stream"))]` if `source_url` is set, because `file` and `file_path` should be None.

Reply:
Done.